ui/raidboss/data/06-ew/trial/zodiark-ex.ts

```diff
@@ -19,6 +192,401 @@ const triggerSet: TriggerSet<Data> = {
+          }
+        }
+      },
+      alertText: (data, matches, output) => {
+        const target = data.activeSigils[data.activeSigils.length - 1];
+        if (!target) {
+          console.error(`Red Tether: Missing target, ${JSON.stringify(matches)}`);
+          return;
+        }
+
+        if (target.x < 100)
+          return output.east!();
+
+        if (target.x > 100)
+          return output.west!();
+
+        if (target.y < 100)
+          return output.south!();
+        return output.north!();
+      },
+      outputStrings: directionOutputStrings,
+    },
+    {
+      id: 'ZodiarkEx Roiling Darkness Spawn',
+      type: 'AddedCombatant',
+      netRegex: NetRegexes.addedCombatant({ name: 'Roiling Darkness', capture: false }),
+      suppressSeconds: 1,
+      response: Responses.killAdds(),
+    },
+    {
+      // 67E4 Green Beam, 67E5 Rectangle, 67E6 Wedge
+      id: 'ZodiarkEx Arcane Sigil Start',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['67E4', '67E5', '67E6'], source: 'Arcane Sigil' }),
+      run: (data, matches, _output) => {
+        if (parseFloat(matches.y) < 100)
+          data.activeFrontSigils.push({ x: parseFloat(matches.x), y: parseFloat(matches.y), typeId: matches.id, npcId: matches.sourceId });
+      },
+    },
+    {
+      id: 'ZodiarkEx Arcane Sigil Start Cleanup',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['67E4', '67E5', '67E6'], source: 'Arcane Sigil', capture: false }),
+      delaySeconds: 0.2,
+      suppressSeconds: 0.2,
+      alertText: (data, _matches, output) => {
+        const activeFrontSigils = data.activeFrontSigils;
+        data.activeFrontSigils = [];
+        if (activeFrontSigils.length === 1 && activeFrontSigils[0]?.typeId === '67E4')
+          return output.sides!();
+        if (activeFrontSigils.length === 2 && activeFrontSigils[0]?.typeId === '67E4' && activeFrontSigils[1]?.typeId === '67E4')
+          return output.middle!();
+        if (activeFrontSigils.length === 3) {
+          for (const sig of activeFrontSigils) {
+            // Find the middle sigil
+            if (sig.x > 90 && sig.x < 110) {
+              if (sig.typeId === '67E4')
+                return output.frontsides!();
+              if (sig.typeId === '67E5')
+                return output.backmiddle!();
+              if (sig.typeId === '67E6')
+                return output.frontmiddle!();
+            }
+          }
+        }
+      },
+      outputStrings: {
+        frontsides: {
+          en: 'front sides',
+        },
+        backmiddle: {
+          en: 'back middle',
+        },
+        frontmiddle: {
+          en: 'front middle',
+        },
+        sides: Outputs.sides,
+        middle: Outputs.middle,
+      },
+    },
+    {
+      id: 'ZodiarkEx Explosion',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: '67E7', source: 'Zodiark' }),
```

Comment:
I think these might be too late to do much about: https://www.twitch.tv/videos/1244188025?t=32m20s.  Are there combatants for these that are placed earlier, somehow?

Reply:
I haven't found anything earlier, something has to draw the star patterns earlier but I didn't see anything in the log for that. Director maybe? But even if it doesn't seem to have the information I need.

Reply:
Yeah, I think it's just not in the logs.  It's in the network data somewhere (surely), but just not anything that makes it to the ACT log.  Maybe some day in the future we'll have some way to emit lines from the network for stuff like this, but not yet.